apps/aechannel/src/aesc_fsm.erl

```diff
@@ -420,25 +420,48 @@ set_ongoing(D) ->
 
 initiate(Host, Port, #{} = Opts0) ->
     lager:debug("initiate(~p, ~p, ~p)", [Host, Port, Opts0]),
-    Opts = maps:merge(#{client => self()}, Opts0),
-    case check_amounts(Opts) of
+    Opts = maps:merge(#{client => self(),
+                        role   => initiator}, Opts0),
+    case init_checks(Opts) of
         ok ->
             start_link(#{ host => Host
                         , port => Port
-                        , opts => Opts#{role => initiator} });
+                        , opts => Opts });
         {error, _Reason} = Err -> Err
     end.
 
 respond(Port, #{} = Opts0) ->
     lager:debug("respond(~p, ~p)", [Port, Opts0]),
-    Opts = maps:merge(#{client => self()}, Opts0),
-    case check_amounts(Opts) of
+    Opts = maps:merge(#{client => self(),
+                        role   => responder}, Opts0),
+    case init_checks(Opts) of
         ok ->
             start_link(#{ port => Port
-                        , opts => Opts#{role => responder} });
+                        , opts => Opts });
         {error, _Reason} = Err -> Err
     end.
 
+init_checks(Opts) ->
+    #{initiator   := Initiator,
+      responder   := Responder,
+      role        := Role,
+      lock_period := LockPeriod} = Opts,
+    Checks = [fun() -> check_amounts(Opts) end,
+              fun() -> check_accounts(Initiator, Responder, Role) end,
+              fun() ->
+                  case LockPeriod < 0 of
```

Comment:
Is there any case in which a Lock Period of 0 makes sense?

Reply:
Although not advised to use `LockPeriod = 0` on main net, there is no actual reason to forbid it. We've used it in some examples aready so people get _fast_ channel opening, for example on testnet.